style: apply /style-guide pass to models/artifacts (#2717)
wandb/docs: models/artifacts/artifacts-walkthrough.mdx

title: "Tutorial: Create, track, and use a dataset artifact"
---

> This walkthrough demonstrates how to create, track, and use a dataset artifact with W&B. By the end, you've logged a dataset as a versioned artifact to W&B and downloaded it in a subsequent run. This lets you reproducibly share datasets across experiments and track them as inputs and outputs of your runs.

Reviewer: Make task-oriented

```suggestion
This walkthrough shows you how to create, track, and use a dataset artifact with W&B. You log a dataset as a versioned artifact to W&B, then download it in a later run. This workflow helps you reproducibly share datasets across experiments and track them as inputs and outputs of your runs.
```

Author: Applied your task-oriented intro verbatim.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## Log in to W&B

Import the W&B library and log in to W&B. If you haven't done so already, sign up for a free W&B account.

```python
import wandb

wandb.login()
```

## Initialize a run

Use [`wandb.init()`](/models/ref/python/functions/init) to initialize a run. This generates a background process to sync and log data. Provide a project name and a job type:

```python
# Create a W&B Run. Here you specify 'dataset' as the job type since this example
# shows how to create a dataset artifact.
with wandb.init(project="artifacts-example", job_type="upload-dataset") as run:
    # Your code here
```

## Create an artifact object

Create an artifact object with [`wandb.Artifact()`](/models/ref/python/experiments/artifact). Provide a name for the artifact and a description of the file type for the `name` and `type` parameters, respectively.

> For example, the following code snippet demonstrates how to create an artifact called `'bicycle-dataset'` with a `'dataset'` label:

Reviewer: ```suggestion
For example, the following code snippet demonstrates how to create an artifact called `bicycle-dataset` with a `dataset` label:
```

Author: Applied — removed the inner quotes so it reads `bicycle-dataset` and `dataset`.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

Replace the filename `dataset.h5` in the previous code snippet with the path to the file you want to add to the artifact.

## Log the dataset

> Use the W&B run object's `wandb.Run.log_artifact()` method to both save your artifact version and declare the artifact as an [output of the run](/models/artifacts/explore-and-traverse-an-artifact-graph).

Reviewer: ```suggestion
Use the run object's `wandb.Run.log_artifact()` method to both save your artifact version and declare the artifact as an [output of the run](/models/artifacts/explore-and-traverse-an-artifact-graph).
```

Author: Applied — dropped "W&B" so it now reads "Use the run object's …".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

```python
# Save the artifact version to W&B and mark it
# as the output of this run
run.log_artifact(artifact)
```

> When you log an artifact, W&B creates a `'latest'` [alias](/models/artifacts/create-a-custom-alias) by default. For more information about artifact aliases and versions, see [Create a custom alias](./create-a-custom-alias) and [Create new artifact versions](./create-a-new-artifact-version), respectively.

Reviewer: ```suggestion
When you log an artifact, W&B creates a `latest` [alias](/models/artifacts/create-a-custom-alias) by default. For more information about artifact aliases and versions, see [Create a custom alias](./create-a-custom-alias) and [Create new artifact versions](./create-a-new-artifact-version), respectively.
```

Author: Applied your suggestion, including `latest` without quotes and the aliases/versions cross-references.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

```

## Download and use the artifact

> Now that the dataset is logged as an artifact, you can pull it into other runs as a tracked input. The following code example demonstrates the steps you can take to use an artifact you've logged and saved to the W&B servers:

Reviewer: This means nothing, confusing "....pull it into other runs as a tracked input."  Remove.

Author: Removed the confusing "pull it into other runs as a tracked input" sentence.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

  artifact_dir = artifact.download()
```

Alternatively, you can use the Public API (`wandb.Api`) to export or update data already saved in W&B outside of a run. For more information, see [Track external files](./track-external-files).

> You now have a versioned dataset artifact logged to W&B and consumed by a downstream run. The artifact graph tracks both the upload and the download.

Reviewer: There is not such thing as an artifact graph. There are lineage graphs. However, that is never referenced in this doc so this comes out of no where.

```suggestion
You now have a versioned dataset artifact logged to W&B and consumed by a downstream run. 
```

Author: Removed the made-up "artifact graph" sentence; the closing line now matches your suggested wording.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
